lib/src/types.dart

```diff
@@ -159,3 +165,19 @@ class VideoDimensions {
         height ?? this.height,
       );
 }
+
+@immutable
+class ParticipantTrackPermission {
+  /// The participant id this permission applies to.
+  final String participantSid;
+
+  // The list of track ids that the target participant can subscribe to.
+  // When unset, it'll allow all tracks to be subscribed by the participant.
+  // When empty, this participant is disallowed from subscribing to any tracks.
+  final List<String>? allowedTrackSids;
```

Comment:
Hmm, should probably just keep the `allTracksAllowed` allowed flag rather than this unset vs empty behavior difference. It'll keep the behavior similar across platforms.

Reply:
This behavior was copied from the JS SDK PR. I agree it should be similar across platforms.
https://github.com/livekit/client-sdk-js/pull/109/files#diff-0cccdd41433a8b9763e61ed69f5e048472997700549df11e3b6c6af08d15af1aR9-R14

Maybe I will follow Android's PR 🤔 Would like to hear @davidzhao 's opinion.

Reply:
changed it!

Reply:
I think DL makes a fair point about consistency. With JS I mainly removed it because it was quite verbose setting that field when testing. Since JS uses an interface, the user has to explicitly type out all these fields.

With that said, I agree consistency + clarity outweighs verbosity, and will change it in JS as well.